### ISSUE TITLE
Non-webkit browsers ignore bottom padding for scrolling <hx-div>

### ISSUE THREAD

**Issue by CITguy** [rackerlabs/helix-ui#379]
## Describe the Bug
Bottom margin on scrolling containers is ignored in non-webkit browsers (Firefox, Edge, and IE).

## Steps to Reproduce
1. Go to https://rackerlabs.github.io/helix-ui/components/box/#scrolling-containers
2. Scroll to the bottom of the container

## Expected behavior
Scrolling content should remain above the bottom padding of the `<hx-div>` container, when scrolled to the bottom.

## Screenshots
👍 _Chrome 70.0.3538.77: Scrolling content stays above the bottom padding_
<img width="696" alt="screen shot 2018-11-16 at 11 19 21 pm" src="https://user-images.githubusercontent.com/545605/48657439-4abe1d00-e9f6-11e8-83e8-630df4727c84.png">

👍 _Safari 12.0.1: Scrolling content stays above the bottom padding_
<img width="697" alt="screen shot 2018-11-16 at 11 20 34 pm" src="https://user-images.githubusercontent.com/545605/48657446-7f31d900-e9f6-11e8-849c-d7b53556a955.png">

👎 _Firefox 63.0.3: Scrolling content does not remain above bottom padding_
<img width="752" alt="screen shot 2018-11-16 at 11 19 55 pm" src="https://user-images.githubusercontent.com/545605/48657454-95d83000-e9f6-11e8-96ee-45486f951acf.png">

👎 _Edge 17.17134: Scrolling content does not remain above bottom padding_
![edge 17 margin collapse bug](https://user-images.githubusercontent.com/545605/48657555-3549f280-e9f8-11e8-9001-3fd33dc857f2.png)

👎 _IE11: Scrolling content does not remain above bottom padding_
![ie11 margin collapse bug](https://user-images.githubusercontent.com/545605/48657556-3549f280-e9f8-11e8-9b33-3e2b5ce29ab6.png)

## Environment
Please complete the following information:

 - Device: Desktop
 - OS: MacOS, Windows 10
 - Browser: Firefox, Edge, IE11

## Additional Context
This is suspected to be an incorrect browser implementation of margin collapsing. According to [spec](https://www.w3.org/TR/CSS2/box.html#collapsing-margins), margins should only collapse if they actually touch.  However, the bottom margin of the content (even if 0) never meets the margin of the container, because there is padding between the two, so margin collapsing should not occur.

**Comment by CITguy** [comment]
TL;DR - CSSWG hasn't fully [finalized the spec](https://github.com/w3c/csswg-drafts/issues/129) on how it _should_ behave and browser vendors are just beginning to discuss how to update.

Bugs have been [filed for over 7 years](https://bugzilla.mozilla.org/show_bug.cgi?id=748518) on this issue.  Technically, Firefox, IE and Edge are to spec, but web developers expect the padding to be preserved (which is how Chrome, Safari, and Opera have chosen to implement it).

Based on "recent" developments ([April 2018](https://github.com/w3c/csswg-drafts/issues/129#issuecomment-380815931)), the CSSWG seems to have agreed on clarification to the spec at least in the block-axis, but are still [discussing behavior in the inline-axis](https://github.com/w3c/csswg-drafts/issues/129#issuecomment-417525242). If I interpret that correctly, this means that we'll probably see Edge and Firefox update to align with the expected behavior of preserving the padding in the block-axis (same as Chrome, Safari, and Opera). We may have to wait it out for a couple of releases of Firefox and Edge to see what happens.  However, chances are slim to none for any update to IE11, so we _might_ need to brainstorm on another solution (if we still plan on continuing support for IE).

**Comment by CITguy** [comment]
The ultimate issue is that you can't reliably keep the padding if it's on the same container that scrolls. To fix this, we'll have to update `<hx-div>` to provide a shadow dom wrapper around light dom content for purposes of padding.
